File: execution-worker/docker/sandbox-python/entrypoint.py
# Same role as the JavaScript sandbox's entrypoint.js: read the assembled
# payload (user code + inline test harness, built by python.runner.ts) from
# stdin and exec it. The container boundary — not this script — is what
# actually provides isolation.
import sys

source = sys.stdin.read()
try:
    exec(source, {"__name__": "__main__"})
except Exception as e:
    print(str(e), file=sys.stderr)
    sys.exit(1)

File: execution-worker/docker/sandbox-javascript/entrypoint.js
// Reads the full job payload (user code + test harness, already assembled by
// javascript.runner.ts) from stdin, evaluates it, and lets it print its own
// single JSON results line to stdout. This file intentionally does almost
// nothing — the isolation comes from the container boundary (§5 of the
// architecture spec: no network, memory/CPU/pid caps, read-only filesystem,
// wall-clock timeout enforced by the worker), not from anything clever here.

let input = "";
process.stdin.on("data", (chunk) => { input += chunk; });
process.stdin.on("end", () => {
  try {
    // eslint-disable-next-line no-eval
    eval(input);
  } catch (err) {
    console.error(String(err));
    process.exit(1);
  }
});
